### PR TITLE
Fix languagetool findings

### PR DESCRIPTION
**Pull request by florian-richter**
## What Has Changed?

I recently found https://github.com/davidlday/vscode-languagetool-linter which is like grammarly but offline and open-source.
I went through the docs and applied all fixes that did not require bigger rewrites (missing commas, spelling, ...).
I found some more small fixes while going through the docs.

The meaning should be identical everywhere but many small (and some larger) errors should be fixed now.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)

### CODE REVIEW

File: docs/java/features/odata/generic-odata-client.mdx [hunk 1]
```diff
@@ -68,10 +68,12 @@ ODataRequestAction
 ```
 
 :::note
+
 What used to be function imports in OData V2 have been split into functions and actions in the OData V4 standard.
 The above classes adopt the OData V4 notion.
 To perform V2 function imports leverage the function class in case the operation should use HTTP GET.
-Instead use the action class if the operation should use POST.
+Instead, use the action class if the operation should use POST.
```
Comment (marikaner) [inline docs/java/features/odata/generic-odata-client.mdx:75]:
```suggestion
Instead, if the operation should use POST, use the class for actions.
```

File: docs/js/features/odata/common/batch/changeset.mdx [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+A changeset is a collection of HTTP `POST`, `PUT`, `PATCH` and `DELETE` operations - requests built by any [`CreateRequestBuilder`](#create-request-builder), [`UpdateRequestBuilder`](#update-request-builder) and [`DeleteRequestBuilder`](#delete-request-builder) in terms of the SAP Cloud SDK.
 The order of execution within a changeset is not defined.
 This differs from the whole batch request itself, where the order is defined.
-Therefore the requests within a changeset should not depend on each other.
-If the execution of any of the requests within a changeset fails, the whole changeset will be reflected as an error in the response and will not be applied, much like a database transaction.
+Therefore, the requests within a changeset should not depend on each other.
+If the execution of any request within a changeset fails, the whole changeset will be reflected as an error in the response and will not be applied, much like a database transaction.
```
Comment (marikaner) [inline docs/js/features/odata/common/batch/changeset.mdx:5]:
```suggestion
If the execution of any request within a changeset fails, the whole changeset will be reflected as an error in the response.
The changeset will not be applied, much like a database transaction.
```

File: docs/js/features/odata/common/entity/from-json.mdx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-In some cases it makes sense for you to store your data as a JSON object, that is based on the entity type, i. e. using the property names of the entity class as properties of your object.
+In some cases it makes sense for you to store your data as a JSON object, that is based on the entity type, i.e. using the property names of the entity class as properties of your object.
```
Comment (marikaner) [inline docs/js/features/odata/common/entity/from-json.mdx:1]:
```suggestion
Sometimes, it makes sense for you to store your data as a JSON object, that is based on the entity type.
For example, when using the property names of the entity class as properties of your object.
```

File: docs/js/features/odata/common/operations/count.mdx [hunk 4]
```diff
@@ -19,12 +19,10 @@ BusinessPartner.requestBuilder()
 As defined in the OData spec `count` is not affected by `top`, `skip`, and `orderBy`.
 :::note `top()` and `skip()` are ignored for count
 If you include these methods in a count request they will be ignored by the SAP Cloud SDK.
-So these three requests:
+So these three requests will all return the same value.
```
Comment (marikaner) [inline docs/js/features/odata/common/operations/count.mdx:22]:
```suggestion
These three requests will all return the same value.
```

File: docs/js/features/odata/overview.mdx [hunk 5]
```diff
@@ -37,14 +37,14 @@ The SAP Cloud SDK simplifies the consumption of OData services through the follo
...
-- We ship pre-generated OData clients for the SAP S/4HANA OData services.
+- We ship pregenerated OData clients for the SAP S/4HANA OData services.
   They are also referred to as the _Virtual Data Model_ or _VDM_.
   You can find a list of all those services in the [SAP API Business Hub](https://api.sap.com/package/SAPS4HANACloud?section=Artifacts) (check the ODATA box).
 - We support both OData v2 and OData v4 protocols in the SAP Cloud SDK for JavaScript.
 
 ### Code Generator
 
-If you need a client for an OData service where we do not ship a pre-generated client use our versatile [code generator](generate-typed-odata-v2-and-v4-client-for-javascript-and-typescript) to convert OData service definition into the type-safe Java client library.
+If you need a client for an OData service where we do not ship a pregenerated client use our versatile [code generator](generate-typed-odata-v2-and-v4-client-for-javascript-and-typescript) to convert OData service definition into the type-safe Java client library.
```
Comment (marikaner) [inline docs/js/features/odata/overview.mdx:47]:
```suggestion
If you want to use an OData service, that has no pregenerated client, you can use the [client generator](generate-typed-odata-v2-and-v4-client-for-javascript-and-typescript) to generate your own client.
```

File: docs/js/guides/tutorial-overview-sdk-js.md [hunk 6]
```diff
@@ -7,7 +7,7 @@ sidebar_label: Tutorials Overview
 description: This document will give you an overview of the different tutorials for the SAP Cloud SDK for JavaScript.
 ---
 
-To give you a quick and easy start to developing with SAP Cloud SDK for JavaScript we created a set of comprehensive tutorials split into missions and groups for a convenient learning path.
+To give you a quick and easy start to developing with the SAP Cloud SDK for JavaScript, we created a set of comprehensive tutorials split into missions and groups for a convenient learning path.
```
Comment (marikaner) [inline docs/js/guides/tutorial-overview-sdk-js.md:10]:
```suggestion
To give you a quick and easy start to developing with the SAP Cloud SDK for JavaScript, we created a set of comprehensive tutorials.
```

File: docs/js/release-notes.mdx [hunk 7]
```diff
@@ -28,24 +28,17 @@ It will help you:
...
 :::
 
-<!-- vale SAP.Products = YES -->
-
 ## Pregenerated Type-Safe Client Libraries (VDM)
 
 :::tip What are client libraries?
 
-For your convenience we [pre-generate type-safe client libraries](https://www.npmjs.com/search?q=%40sap%2Fcloud-sdk-vdm-*) (also known as Virtual Data Model or VDM for short) for allow-listed OData services of SAP S/4HANA Cloud, Marketing Cloud and SAP S/4HANA On-premise.
+For your convenience we [pregenerate type-safe client libraries](https://www.npmjs.com/search?q=%40sap%2Fcloud-sdk-vdm-*) (also known as Virtual Data Model or VDM for short) for allow-listed OData services of SAP S/4HANA Cloud, Marketing Cloud and SAP S/4HANA On-premise.
```
Comment (marikaner) [inline docs/js/release-notes.mdx:41]:
```suggestion
For your convenience we [pregenerate client libraries](https://www.npmjs.com/search?q=%40sap%2Fcloud-sdk-vdm-*) (also known as Virtual Data Model or VDM for short) for allow-listed OData services of SAP S/4HANA Cloud, Marketing Cloud and SAP S/4HANA On-premise.
```

**Review by marikaner — APPROVED** [review]
I started reading everything, but this is a bit too much. Instead I made minor improvements where there were notifications from GH ;)

Feel free to adjust.

**Comment by artemkovalyov** [comment]
Funnily enough, Grammarly and Languagetools disagree about how to write pre[-]generated:)

**Pull request merged**